documentation/sdkautomation/sdk_customization.md

```diff
@@ -0,0 +1,762 @@
+    }
+  },
+  "required": [
+    "specFolder", "headSha", "headRef", "repoGitUrl", "repoHttpsUrl",
+    "trigger", "changedFiles", "relatedReadmeMdFiles"
+  ]
+}
+```
+
+### GenerateOutput
+
+Output file for generate script.
+
+#### GenerateOutput Example
+
+```jsonc
+{
+  "packages": [
+    {
+      "packageName": "Microsoft.Cdn",
+      "path": [
+        "sdk/cdn"
+      ],
+      "changelog": {
+        "content": "Feature: something \n Breaking Changes: something\n",
+        "hasBreakingChange": true
+      },
+      "artifacts": [
+        "sdk/cdn/cdn.nuget",
+        "sdk/cdn/cdn.snuget"
+      ],
+      "installInstructions": {
+        "full": "To install something...",
+        "lite": "dotnet something"
+      },
+      "result": "success"
+    }
+  ]
+}
+```
+
+#### GenerateOutput Schema
+
+```jsonc
+{
+  "type": "object",
+  "properties": {
+    "packages": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/PackageResult"
+      }
+    }
+  },
+  "required": [
+    "packages"
+  ],
+  "definitions": {
+    "PackageResult": {
+      "properties": {
+        "packageName": {
+          // Name of package. Will be used in branch name and PR title.
+          // By default it's folder name of first entry in path.
+          "type": "string",
+        },
+        "path": {
+          // List of package content paths.
+          // If the path points to a folder then
+          //   all the content under the folder will be included.
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "changelog": {
+          "type": "object",
+          "properties": {
+            "content": {
+              // Content of changelog in markdown
+              "type": "string"
+            },
+            "hasBreakingChange": {
+              // Does the new package has breaking change
+              "type": "boolean"
+            }
+          },
+          "required": [
+            "content"
+          ]
+        },
+        "artifacts": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "installInstructions": {
+          "type": "object",
+          "properties": {
+            "full": {
+              "type": "string"
+            },
+            "lite": {
+              "type": "string"
+            }
+          },
+          "required": [
+            "full"
+          ]
+        },
+      },
+      "required": [
+        "path"
+      ]
+    }
+  }
+}
+```
+
+### InstallInstructionScriptInput
+
+Input of install instruction script.
+
+#### InstallInstructionScriptInput Example
+
+```jsonc
+{
+  "packageName": "Microsoft.Cdn",
+  "artifacts": [
+    "sdk/cdn/cdn.nuget",
+    "sdk/cdn/cdn.snuget"
+  ],
+  "isPublic": true,
+  "downloadUrlPrefix": "https://portal.azure-devex-tools.com/api/sdk-dl-pub?p=Azure/azure-rest-api-specs/1234/azure-sdk-for-net/",
```

Comment:
I'd like to see if we can use template model here for the install instructions instead of needing to provide exact links to the downloads. Perhaps there are reasons but I'd like to understand that better as we start to test these out.

Reply:
What do you mean by template mode? I'm afraid there would be some advanced logic such as parsing content of artifact, so I leave it as a script.

Reply:
I will have to see how this works in practice but by template model I mean we provide a string that has template variables for the package name that the automation can fill-in with the full links on the back-end.

Reply:
In that case we need to provide a template execution engine which is a little bit complicated (artifacts are array) so it would be better to render it on sdk side or maybe you can share some code in eng/common.